### PR TITLE
South Africa Class 1 - Ryan Smith - Js Core 3 - Week 2

### PR DESCRIPTION
**Pull request by RyanS-Cpt**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

**Review by DouglasVDM — COMMENTED** [review]
Good job storing the API address in a variable.

**Comment by RyanS-Cpt** [comment]
Thanks Douglas

File: challenges/dog-photo-gallery/index.html [hunk 1]
```diff
@@ -0,0 +1,95 @@
+<!DOCTYPE html>
```
Comment (mulindierick) [inline challenges/dog-photo-gallery/index.html:1]: Hi Ryan, nice job brother. 
I love the way you used a different approach for the 'dog photo' generator. I actually used one button as I thought the second button might have been a typo from the instructor. Great job though

**Review by DouglasVDM — COMMENTED** [review]
Good, clean, easy to read code.
Where are you using lines 22 and 23 the "adImg" and "showImg"?

File: challenges/dog-photo-gallery/index.html (hunk 1 above)
Comment (RyanS-Cpt) [reply challenges/dog-photo-gallery/index.html:1]: Thanks Erick, just improvised

**Comment by RyanS-Cpt** [comment]
> Good, clean, easy to read code.
> Where are you using lines 22 and 23 the "adImg" and "showImg"?

Hey Douglas, I was trying to incorporate two buttons as per the readMe to see if I could make it work somehow.

File: challenges/dog-photo-gallery/index.html [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+            else{
+                throw `Error ${response.status}:${response.text}`; 
+            }})
+
+        .then(data => imgArr.push(data.message)) //push image url to array
+        .catch(error => alert(error))
+        });
+
+        function addNewImage(imgUrl){
```
Comment (sirpfaira) [inline challenges/dog-photo-gallery/index.html:77]: Nicely done

**Comment by LawrenceBaatjies** [comment]
HI Ryan Nicely done nesting your fetch method in a function,

**Comment by douglas86** [comment]
There seems to be a problem with dog photo gallery, there is no image been rendered to the display, The geetings app works fine. Consider using prittier for your code, it makes it more readable.

**Comment by RyanS-Cpt** [comment]
> There seems to be a problem with dog photo gallery, there is no image been rendered to the display, The geetings app works fine. Consider using prittier for your code, it makes it more readable.

Hey Douglas it seems to be working on my side, what happens when you run it?

**Pull request closed without merging**